jupyterlab_git/git.py

```diff
@@ -249,20 +249,24 @@ def __init__(self, contents_manager):
             }
 
         result = []
-        line_array = my_output.splitlines()
-        for line in line_array:
-            to1 = None
-            from_path = line[3:]
-            if line[0] == "R":
-                to0 = line[3:].split(" -> ")
-                to1 = to0[len(to0) - 1]
+        line_iterable = iter(my_output.strip("\x00").split('\x00'))
+        result = []
+        for line in line_iterable:
+            x = line[0]
+            y = line[1]
+            if line[0]=='R':
+                #If file was renamed then we need both this line
+                #and the next line, then we want to move onto the subsequent
+                #line. We can accomplish this by calling next on the iterable
+                to = line[3:]
+                from_path = next(line_iterable)
             else:
-                to1 = line[3:]
-            if to1.startswith('"'):
-                to1 = to1[1:]
-            if to1.endswith('"'):
-                to1 = to1[:-1]
-            result.append({"x": line[0], "y": line[1], "to": to1, "from": from_path})
+                #to and from_path are the same
+                from_path = line[3:]
+                to = line[3:]
+            result.append({"x": x, "y": y, "to": to, "from": from_path})
```

Comment:
Based on my understanding on the flow here, this whole chunk inside the for loop could be simplified to:
```
            from_path = line[3:]
            if line[0]=='R':
                #If file was renamed then we need both this line
                #and the next line, then we want to move onto the subsequent
                #line. We can accomplish this by calling next on the iterable
                from_path = next(line_iterable)
            result.append({"x": line[0], "y": line[1], "to": line[3:], "from": from_path})
```

Reply:
Good simplification, I changed to be this.